front/src/modules/trainschedule/components/ImportTrainSchedule/ImportTrainScheduleConfig.tsx

```diff
@@ -203,13 +199,36 @@ const ImportTrainScheduleConfig = ({
         const formattedArrivalTime = `${startDate} ${arrivalTime}`;
         const formattedDepartureTime = `${startDate} ${departureTime}`;
 
+        let stopFor: number | undefined;
+
+        const arrivalDate = new Date(`${startDate}T${arrivalTime}`);
+        const departureDate = new Date(`${startDate}T${departureTime}`);
+        if (ocpType === 'stop') {
+          if (arrivalTime && departureTime) {
+            stopFor = Math.round((departureDate.getTime() - arrivalDate.getTime()) / 1000);
+          } else {
+            stopFor = 0;
+          }
+        } else if (ocpType === 'pass') {
+          if (arrivalTime && departureTime && arrivalTime !== departureTime) {
+            // Incorrect data, it shouldn't happen on a ocpType pass, skip step
+            return null;
+          }
+          if (isLastOcp) {
+            stopFor = 0;
+          } else {
+            stopFor = undefined;
```

Comment:
Optional nit: `undefined` is the initial value of `stopFor` so this isn't strictly needed (but I don't mind).